packages/memento/test/util.test.js

```diff
@@ -0,0 +1,215 @@
+/* eslint-disable no-undef */
+import { expect } from 'chai';
+import { requestWithPagination, handleRateLimit } from '../src/util';
+import { mockEntriesResponse, mockRateLimitExceeded } from './helpers';
+import { enableMockClient } from '@openfn/language-common/util';
+
+const testServer = enableMockClient('https://util.mementodatabase.com');
+const state = {
+  configuration: {
+    baseUrl: 'https://util.mementodatabase.com',
+    token: 'user-api-token',
+  },
+};
+
+describe('handleRateLimit', () => {
+  let originalConsoleLog;
+  let consoleLogCalls;
+
+  beforeEach(() => {
+    consoleLogCalls = [];
+    originalConsoleLog = console.log;
+    console.log = msg => consoleLogCalls.push(msg);
+  });
+
+  afterEach(() => {
+    console.log = originalConsoleLog;
+  });
+
+  const requestConfig = {
+    throttleTime: 1000, // 1s
+    maxRequests: 5,
+  };
+
+  it('should return immediately when no previous requests exist', () => {
```

Comment:
I am grateful for the test - this is a bit of a weird behaviour!  `handleRateLimit` must be declared an `async` function, so therefore always returns a promise

Reply:
Thank you for that catch